tests/MongoDB.Driver.Tests/Specifications/crud/CrudTestRunner.cs

```diff
@@ -270,6 +283,12 @@ private void SkipTestIfNeeded(BsonDocument definition, BsonDocument test)
             {
                 throw new SkipException(reason.AsString);
             }
+
+            if (definition["_path"].AsString.EndsWith("aggregate-out-readConcern.json") &&
```

Comment:
That's in a different test runner.

In this test runner there is already a method called SkipTestIfNeeded and it seemed natural to add any logic for skipping tests to the existing method.

Reply:
I thought about this method like a place which is responsible for `common/high level` `skipping` logic(based on spec json nodes). I'm not sure that a skip reason which we have here should be also processed via `common` logic. 
Let's consider this example, if we finally have a single approach for test runner logic (for instance having a single base class for all test runners), we will be able to simply remove this method and call a base method instead, probably even from a base class. For the current case, we will need to implement some logic to call base class + your current logic.

On the other hand, we have a class `TestCaseFactory` which looks like the best place to decide which tests exactly should be provided for the current test runner.

I guess, it's better to try to use as more similar approaches as possible. We have a test factory for each of our test runners, so, next time when we will have such situation, I guess it's better if we don't think whether we have some `skipping` logic somewhere or not, we will just check a single place which is a factory. Otherwise, we will have two different implementations for the same logic in different runners which can be confused..

Reply:
Sorry for big text

Reply:
Your suggestions can be implemented when we refactor our JSON runners to use common code.

This is just a quick implementation that is appropriate for this class as it currently exists.